### PR TITLE
add FastTextMultiOutputClassifier

### LINKED ISSUE #1: Support multilabel classification
This would be useful for me personally, and I think a nice addition to this awesome package.

A natural approach would be to follow the example of sklearn's [MultiOutputClassifier](https://github.com/scikit-learn/scikit-learn/blob/364c77e04/sklearn/multioutput.py#L352).
- `FastTextMultiOutputClassifer` would inherit from `FastTextClassifier`
- Changes to `FastTextMultiOutputClassifer.fit`
    - Take `Y` as an array-like of shape (n_samples, n_classes)
    - Take label names as a list-like of length n_classes
    - Write the temp file using the multilabel format expected by FastText, e.g. `__label__sauce __label__cheese How much does potato starch affect a cheese sauce recipe?`
- `loss` would be fixed at `ova`
- I think `predict_proba` wouldn't need any changes, since FastText returns a sorted list of label probabilities either way.

If this sounds reasonable, I'm happy to write it.

### PR DESCRIPTION
**Pull request by sandypreiss**
Closes https://github.com/pmbaumgartner/fasttext-lite/issues/1

### CODE REVIEW

File: src/fasttext_lite/core.py [hunk 1]
```diff
@@ -217,3 +223,108 @@ def _get_original_label(self, adjusted_label: str):
...
+            neg=neg,
+            wordNgrams=wordNgrams,
+            loss="ova",
+            bucket=bucket,
+            lrUpdateRate=lrUpdateRate,
+            t=t,
+            label=label,
+            verbose=verbose,
+            thread=thread,
+        )
```
Comment (sandypreiss) [inline src/fasttext_lite/core.py]: This is very un-DRY, but sklearn requires all parameters to be explicit in the `__init__` method's signature for estimators. This is to ensure consistent behavior with various functions that rely on explicit attribute setting, such as `get_params`. So using `**kwargs` throws a runtime error. Still, maybe there is a better way to do this.

Comment (pmbaumgartner) [reply src/fasttext_lite/core.py]: If there's a better way, I haven't found it!

File: src/fasttext_lite/core.py [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-from typing import Dict, List, Union
-
 from tempfile import NamedTemporaryFile
-from typing import Literal
+from typing import Dict, List, Literal, Union
+
+import fasttext
 import numpy as np
 from sklearn.base import BaseEstimator, ClassifierMixin
 from sklearn.exceptions import NotFittedError
```
Comment (pmbaumgartner) [inline src/fasttext_lite/core.py]: ty for fixing these

**Review by pmbaumgartner — APPROVED** [review]

**Comment by pmbaumgartner** [comment]
@sandypreiss check the `mypy` error. I think you might just want to create your own class instead of inheriting from `FastTextClassifier`, and just inherit from the same things that `FastTextClassifier` does.

**Comment by sandypreiss** [comment]
This has been a nice OOP lesson. I think an abstract base class is the best solution here. Lmk what you think.

File: src/fasttext_lite/core.py [hunk 3]
```diff
@@ -37,8 +37,8 @@ def convert_path(path: StrOrPath) -> Path:
 
 class BaseFastTextClassifier(ABC, BaseEstimator, ClassifierMixin):
     @abstractmethod
-    def fit(self):
-        pass
+    def fit(self, X, y) -> "BaseFastTextClassifier":
+        return self
```
Comment (sandypreiss) [inline src/fasttext_lite/core.py:41]: Per further reading, matching the sklearn convention of fit returning self, to enable method chaining like `model.fit().predict()`

File: src/fasttext_lite/core.py [hunk 4]
```diff
@@ -274,20 +277,20 @@ def __init__(
         self.is_quantized = False
         self.adjusted_labels: Dict[str, str] = {}
 
-    def fit(self, X, Y, labels) -> None:
+    def fit(self, X, y) -> "FastTextMultiOutputClassifier":
```
Comment (sandypreiss) [inline src/fasttext_lite/core.py:280]: Moving labels to init allows the signature to match across both classifiers and the base class.

**Comment by sandypreiss** [comment]
@pmbaumgartner thoughts on this iteration? I didn't implement your suggestion to inherit from sklearn separately yet. I wanted to resolve the mypy error while sticking with this design (mostly as an exercise to distinguish the issue causing the mypy error from the question of how best to align with sklearn's design).

No worries if you still prefer inheriting from sklearn separately. I just want to make sure your opinion hasn't changed now that this version is working.
